### PR TITLE
postgres: add defaultSettings internal option

### LINKED ISSUE #30: postgres: provide a way to add to default settings instead of simply overriding all of them
the following defaults: 

https://github.com/juspay/services-flake/blob/f3a7ea9ec9166f8b97e0f303668d3e0608835e2b/nix/postgres.nix#L139-L143

are needed for the setup to function correctly and would rarely need to be overriden

a user however would often want to add more settings

in my case I'd like to add `shared_preload_libraries = 'timescaledb'`

we could add `extraSettings` which would be merged with `settings` or move current defaults to `defaultSettings` and leave `settings` empty, for the user to set

### PR DESCRIPTION
**Pull request by shivaraj-bh**
closes #30

### CODE REVIEW

File: nix/postgres.nix [hunk 1]
```diff
@@ -136,6 +136,22 @@ in
...
+            type = lib.types.raw;
+            internal = true;
+            readonly = true;
+            description = ''
+              Default configuration for `postgresql.conf`. `settings` can override these values.
+            '';
+            default = {
+              listen_addresses = config.listen_addresses;
+              port = config.port;
+              unix_socket_directories = lib.mkDefault config.dataDir;
```
Comment (srid) [inline nix/postgres.nix:150]: Is `lib.mkDefault` necessary here?

File: nix/postgres.nix [hunk 2]
```diff
@@ -136,6 +136,22 @@ in
           '';
         };
 
+        defaultSettings =
+          lib.mkOption {
+            type = lib.types.raw;
```
Comment (srid) [inline nix/postgres.nix:141]: Shouldn't this be `attrsOf (oneOf [ bool float int str ]);`?

**Review by srid — APPROVED** [review]
LGTM (aside from possibly the `type`)

File: nix/postgres.nix (hunk 1 above)
Comment (shivaraj-bh) [reply nix/postgres.nix]: removed it! We don't care much about the priority of the default here anyway.

File: nix/postgres.nix (hunk 2 above)
Comment (shivaraj-bh) [reply nix/postgres.nix]: Since it was `internal` I thought it's fine to be raw, but doesn't hurt to be more specific.

**Pull request merged**